### PR TITLE
POC to have curl linked in windows

### PR DESCRIPTION
**Pull request by vhvb1989**

### CODE REVIEW

File: sdk/core/core/src/main.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (sergey-shandar) [inline sdk/core/core/src/main.c]: Should we move this file to `test` folder?

Comment (vhvb1989) [reply sdk/core/core/src/main.c]: yeah, what I want is to see the Pipeline working for Windows check.. But I don't see checks coming up

Comment (vhvb1989) [reply sdk/core/core/src/main.c]: Update: move the curl easy perform to test.
We can remove it later. For now it is a simple example on how to use curl.  CI is working for all platforms now.  On windows, it works on Visual Studio

**Comment by vhvb1989** [comment]
> We probably should keep -Werror, and avoid/disable the warnings that would appear.

Yes, I have reverted this and let the Warning only...  @sergey-shandar  can you take care of that Warning in the parser?  either disable it or updating code?

**Comment by antkmsft** [comment]
@vhvb1989, was it the -Wimplicit-fallthrough warning? The one in the switch/case statement?

**Comment by antkmsft** [comment]
https://github.com/Azure/azure-sdk-for-c/pull/71

**Pull request merged**